metis-indexing/src/main/java/eu/europeana/indexing/IndexerPool.java

```diff
@@ -0,0 +1,166 @@
+ * they are needed (i.e. when all other indexers are in use already). This thread pool implements
+ * the automatic destruction of idle indexers, given a certain permissible idle time.</p>
+ * <p>This class is thread-safe.
+ * </p>
+ */
+public class IndexerPool implements Closeable {
+
+  private final GenericObjectPool<Indexer> pool;
+
+  /**
+   * Constructor.
+   *
+   * @param indexingSettings The  settings with which to create the indexer instances in the pool.
+   * @param maxIdleTimeForIndexerInSecs The idle time after which an indexer is eligible for
+   * destruction, in seconds.
+   * @param idleTimeCheckIntervalInSecs The interval with which we check the idle time of indexers
+   * to decide whether to destroy them, in seconds.
+   */
+  public IndexerPool(IndexingSettings indexingSettings, long maxIdleTimeForIndexerInSecs,
+      long idleTimeCheckIntervalInSecs) {
+    this(new IndexerFactory(indexingSettings), maxIdleTimeForIndexerInSecs,
+        idleTimeCheckIntervalInSecs);
+  }
+
+  /**
+   * Constructor.
+   *
+   * @param indexerFactory The factory from which to create the indexer instances in the pool.
+   * @param maxIdleTimeForIndexerInSecs The idle time after which an indexer is eligible for
+   * destruction, in seconds.
+   * @param idleTimeCheckIntervalInSecs The interval with which we check the idle time of indexers
+   * to decide whether to destroy them, in seconds.
+   */
+  public IndexerPool(IndexerFactory indexerFactory, long maxIdleTimeForIndexerInSecs,
+      long idleTimeCheckIntervalInSecs) {
+
+    // Create indexer pool with default options.
+    pool = new GenericObjectPool<>(new PooledIndexerFactory(indexerFactory));
+
+    // Set custom options for the size of the pool: no max or min number of indexer objects.
+    pool.setMaxIdle(-1);
+    pool.setMinIdle(-1);
+    pool.setMaxTotal(-1);
+
+    // Set custom options for indexer pool regarding eviction (when indexer has been idle for some time).
+    pool.setSoftMinEvictableIdleTimeMillis(-1);
+    pool.setMinEvictableIdleTimeMillis(convertSecsToMillis(maxIdleTimeForIndexerInSecs));
+    pool.setTimeBetweenEvictionRunsMillis(convertSecsToMillis(idleTimeCheckIntervalInSecs));
+  }
+
+  private static long convertSecsToMillis(long seconds) {
+    return Duration.ofSeconds(seconds).toMillis();
+  }
+
+  /**
+   * <p>
+   * This method indexes a single record, using a free indexer in the pool.
+   * </p>
+   *
+   * @param record The record to index (can be parsed to RDF).
+   * @param preserveUpdateAndCreateTimesFromRdf This determines whether this indexer should use the
+   * updated and created times from the incoming RDFs, or whether it computes its own.
+   * @throws IndexingException In case a problem occurred during indexing.
+   * @throws IndexerConfigurationException In case an exception occurred while setting up an
+   * indexer.
+   */
+  public void index(String record, boolean preserveUpdateAndCreateTimesFromRdf)
```

Comment:
Since this is something that is used externally from metis project and now that we have the chance let's have this method and the one below(indexRdf) throw one exception, that would encapsulate both current exception cases. What do you think?

Reply:
Fixed.